packages/react-aria-components/src/FileTrigger.tsx

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2023 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import {AriaLabelingProps} from '@react-types/shared';
+import {ButtonContext} from './Button';
+import {ContextValue, DOMProps, Provider, SlotProps, useContextProps} from './utils';
+import {filterDOMProps} from '@react-aria/utils';
+import {Input} from './Input';
+import {LinkContext} from './Link';
+import React, {createContext, ForwardedRef, forwardRef, useRef} from 'react';
+
+export interface FileTriggerProps extends SlotProps, DOMProps, AriaLabelingProps {
+  /**
+   * Specifies what file types are allowed.
+   */
+  accept?: string,
+  /**
+   * Whether multiple files can be selected.
+   */
+  multiple?: boolean,
+  /**
+   * Handler that is called on submission.
+   */
+  onChange?: React.ChangeEventHandler<HTMLInputElement>
+}
+
+export const FileTriggerContext = createContext<ContextValue<FileTriggerProps, HTMLDivElement>>(null);
+
+function FileTrigger(props: FileTriggerProps, ref: ForwardedRef<HTMLDivElement>) {
+  [props, ref] = useContextProps(props, ref, FileTriggerContext);
+  let {onChange, accept, className, multiple, ...otherProps} = props;
+  let inputRef = useRef<HTMLInputElement>(null);
+
+  return (
+    <Provider
+      values={[
+        [LinkContext, {slot: 'file', onPress: () => inputRef.current?.click()}],  
+        [ButtonContext, {slot: 'file', onPress: () => inputRef.current?.click()}]
+      ]}>
```

Comment:
We could probably use `PressResponder` here rather than specific slots, that way any pressable element can trigger the input. I also don't think we need the `slot: 'file'` here?